monitoring/prometheus/alerts/ceph_default_alerts.yml

```diff
@@ -234,10 +229,9 @@ groups:
       - alert: pool filling up
         expr: |
           (
-            (
-              ceph_pool_max_avail / deriv(ceph_pool_max_avail[2d])
-            ) * on(pool_id) group_right ceph_pool_metadata <= 5
-          ) > 0
+            predict_linear(ceph_pool_stored[2d], 3600 * 24 * 5) >=
+            ceph_pool_max_avail
+          ) * on(pool_id) group_right(name) ceph_pool_metadata
```

Comment:
I think a `group_left` should be used here, otherwise the label `name` is not added to the result (and i think that's the intention of this action).

I modified the query a little bit by removing `>= ceph_pool_max_avail` to get a result, even if the pool is not full.

# group_right
`(predict_linear(ceph_pool_stored[2d], 3600 * 24 * 5)) * on(pool_id) group_right(name) ceph_pool_metadata`
![Bildschirmfoto vom 2020-04-08 11-53-16](https://user-images.githubusercontent.com/1897962/78770920-9aa4d780-798f-11ea-8ef0-908815e9f970.png)

# group_left
`(predict_linear(ceph_pool_stored[2d], 3600 * 24 * 5)) * on(pool_id) group_left(name) ceph_pool_metadata`
![Bildschirmfoto vom 2020-04-08 11-47-54](https://user-images.githubusercontent.com/1897962/78770935-a1334f00-798f-11ea-8fd8-0cc0a38a6ed8.png)

Reply:
yup you are absolutely correct. `group_[left,right]` explicitly tells prometheus which side has the higher cardinality on many to one matchings, _not_ the more naive assumption to denote from which side to takes the labels. :+1:

Reply:
In this match, is it possible for any side to have a higer cardinality?